### DISCUSSION TITLE
APIRouter nesting not working

### DISCUSSION THREAD

**Discussion by GigantPro** [fastapi/fastapi#10149] [Questions] [answered]
I have the following structure:
_app (root)
first_router (included to root)
second_router (included to first_router)_

console output:
```
WARNING: StatReload detected changes in 'filename'. Reloading...
And that's it.
```
When accessing the web documentation, Nginx returns this to the console:
```
"GET /api/docs HTTP/1.1" 504 167 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0"
[error] 20#20: *10 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 172.25.0.1, server: , request: "GET /api/docs HTTP/1.1", upstream: "http://172.25.0.2:5001/docs", host: "localhost"
```

**Comment by JavierSanchezCastro** [comment]
Hi!

Please, provide a basic example to reproduce your error

**Comment by GigantPro** [comment, marked as answer]
Oh, its my bad... I inadvertently include the router in itself... Sorry and thanks a lot!

**Comment by WilliamStam** [comment]
know this is a closed issue and you found the problem. but incase somone else lands up here, im having a strange issue that using pycharm "run" configuration isnt the same as pycharm terminal, run config gets stuck on `detected changes in 'filename'. Reloading...` (this is just a FYI thing incase you (dear reader) didnt include the router in itself but having the Reloading... thing
